pkgs/development/haskell-modules/default.nix

```diff
@@ -7,26 +7,24 @@
 , configurationNix ? import ./configuration-nix.nix
 }:
 
+self: # Provided by `callPackageWithSelf`
+
 let
 
   inherit (stdenv.lib) extends makeExtensible;
   inherit (import ./lib.nix { inherit pkgs; }) overrideCabal makePackageSet;
 
   haskellPackages = makePackageSet {
     package-set = initialPackages;
-    inherit ghc extensible-self;
+    extensible-self = self;
+    inherit ghc;
   };
 
   commonConfiguration = configurationCommon { inherit pkgs; };
   nixConfiguration = configurationNix { inherit pkgs; };
 
-  extensible-self = makeExtensible
-    (extends overrides
-      (extends packageSetConfig
-        (extends compilerConfig
-          (extends commonConfiguration
-            (extends nixConfiguration haskellPackages)))));
-
-in
-
-  extensible-self
+in (extends overrides
+     (extends packageSetConfig
+       (extends compilerConfig
+         (extends commonConfiguration
+           (extends nixConfiguration haskellPackages))))) self
```

Comment:
Are you talking about the `self`? There's no reason it *shouldn't* be the final output, and I didn't want to have to pass around two different `self`s